a10_octavia/common/utils.py

```diff
@@ -58,14 +74,19 @@ def convert_to_rack_vthunder_conf(rack_list):
             validation_flag = validate_params(rack_device)
             if validation_flag:
                 rack_device['undercloud'] = True
+                if 'partition' not in rack_device or not rack_device['partition']:
+                    rack_device['partition'] = a10constants.SHARED_PARTITION
                 vthunder_conf = data_models.VThunder(**rack_device)
+                if rack_dict.get(rack_device['project_id']):
+                    raise exceptions.ValidationException(detail=_('Supplied duplicate project_id ' +
+                                                                  rack_device['project_id'] +
+                                                                  ' in [rack_vthunder]'))
                 rack_dict[rack_device['project_id']] = vthunder_conf
-            else:
-                LOG.warning('Invalid definition of rack device for '
-                            'project ' + rack_device['project_id'])
 
     except KeyError as err:
         LOG.error("Invalid definition of rack device in configuration file."
                   "The Loadbalancer you create shall boot as overcloud."
                   "Check attribute: " + str(err))
-    return rack_dict
+
+    if check_partition_collision(rack_dict):
+        return rack_dict
```

Comment:
Its acceptable IMO, as it doesn't matter what we are returning in case of failure, as it won't move forward once exception is raised due to any kind of misconfiguration in Config. The service won't start.